parsely/src/main/java/com/parsely/parselyandroid/ParselyTracker.java

```diff
@@ -443,8 +444,9 @@ private void enqueueEvent(Map<String, Object> event) {
      * include a call to `flushEventQueue` in your main activity's `onDestroy()` method.
      */
     public void flushEventQueue() {
-        // needed for call from MainActivity
-        new FlushQueue().execute();
+        if (!flushInProgress) {
```

Comment:
Blocker (🚫): Unfortunately, this wouldn't work as is (tested too). What you need is to set `flushInProgress` to `true` right after this `if (!flushInProgress)` check and remove the `flushInProgress = true;` at the beginning of `doInBackground(...)` (not needed), only leaving the `flushInProgress = false;` at the end.